File: 0404-sum-of-left-leaves/0404-sum-of-left-leaves.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int sumOfLeftLeaves(TreeNode root) {
        if(root == null) 
            return 0;
        
        return sumOfLeftLeaves(root, false);
    }
    
    public int sumOfLeftLeaves(TreeNode root, boolean isLeft){
        if(root == null){
            return 0;
        }
        
        if(root.left == null && root.right == null){
            if(isLeft){
                return root.val;
            }
        }
        
        return sumOfLeftLeaves(root.left, true) + sumOfLeftLeaves(root.right, false);                
    }
}

File: Easy/0404-sum-of-left-leaves/0404-sum-of-left-leaves.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var sumOfLeftLeaves = function(root) {
    if(root === null){
        return 0;
    }
        
    return sumLeftNodes(root, false);
};

var sumLeftNodes = function(root, isLeft){
    if(root === null){
        return 0;
    }
    
    if(root.left === null && root.right === null){
        if(isLeft){
            return root.val;
        }
    }
    
    return sumLeftNodes(root.left, true) + sumLeftNodes(root.right, false);    
}

